src/UI/Layout.hs

```diff
@@ -62,12 +79,22 @@ fitLayoutAndAnnotate :: Real a => Size (Maybe a) -> Layout a (Size a) -> ALayout
 fitLayoutAndAnnotate = fitLayoutWith (annotatingBidi layoutAlgebra)
 
 layoutAlgebra :: Real a => Algebra (Fitting (LayoutF a) a) (Maybe (Rect a))
-layoutAlgebra (Cofree (offset, maxSize) runC layout) = case layout of
-  Pure size | maxSize `encloses` size -> Just (Rect offset (fromMaybe <$> size <*> maxSize))
+layoutAlgebra (Cofree (alignment, offset, maxSize) runC layout) = case layout of
+  Pure size | maxSize `encloses` size -> Just $ case alignment of
+    Leading -> Rect offset minSize
+    Trailing -> Rect offset { x = x offset + widthDiff} minSize
+    Centre -> Rect offset { x = x offset + fromIntegral (widthDiff `div'` 2 :: Int)} minSize
+    Full -> Rect offset fullSize
```

Comment:
The current alignment is bubbled up through the context state, alongside the offset and maximum size. Alignment is actually _performed_ only on `Pure` terms, since other terms are always sized relative to their children. In other words, to apply alignment to an entire subtree, it suffices to apply the alignment to the leaves, since the alignment’s effects will bubble up as far as the nearest `Align` term or the root, whichever is sooner.